File: csharp/Original/PublishingHouse/Books/Entities/Author.cs
using PublishingHouse.Core.ValueObjects;

namespace PublishingHouse.Books.Entities;

public record Author(AuthorId Id, AuthorFirstName FirstName, AuthorLastName LastName);
public record AuthorId(Guid Value): NonEmptyGuid(Value);
public record AuthorFirstName(string Value) : NonEmptyString(Value);
public record AuthorLastName(string Value) : NonEmptyString(Value);

File: node.js/src/original/domain/books/entities/author.ts
import { DeepReadonly, NonEmptyString } from '#core/typing';

export type Author = DeepReadonly<{
  authorId: AuthorId;
}>;

export type AuthorId = NonEmptyString<'AuthorId'>;

export type AuthorFirstName = NonEmptyString<'AuthorFirstName'>;

export type AuthorLastName = NonEmptyString<'AuthorLastName'>;
